tests/integration/components/flexberry-dropdown-test.js

```diff
@@ -0,0 +1,612 @@
+  });
+});
+
+test('it renders i18n-ed placeholder', function(assert) {
+  assert.expect(2);
+
+  // Render component.
+  this.render(hbs`{{flexberry-dropdown
+    class=class
+  }}`);
+
+  // Retrieve component.
+  let $component = this.$().children();
+  let $dropdownText = $component.children('div.default.text');
+
+  // Check <dropdown>'s placeholder.
+  assert.strictEqual(
+    Ember.$.trim($dropdownText.text()),
+    Ember.get(I18nRuLocale, 'components.flexberry-dropdown.placeholder'),
+    'Component\'s inner <dropdown>\'s placeholder is equals to it\'s default value from i18n locales/ru/translations');
+
+  // Change current locale to 'en' & check <dropdown>'s placeholder again.
+  this.set('i18n.locale', 'en');
+  assert.strictEqual(
+    Ember.$.trim($dropdownText.text()),
+    Ember.get(I18nEnLocale, 'components.flexberry-dropdown.placeholder'),
+    'Component\'s inner <dropdown>\'s placeholder is equals to it\'s value from i18n locales/en/translations');
+});
+
+test('it renders manually defined placeholder', function(assert) {
+  assert.expect(2);
+
+  // Render component.
+  this.render(hbs`{{flexberry-dropdown
+    placeholder=placeholder
+  }}`);
+
+  // Set <dropdown>'s placeholder' & render component.
+  let placeholder = 'please type some text';
+  this.set('placeholder', placeholder);
+
+  // Retrieve component.
+  let $component = this.$().children();
+  let $dropdownText = $component.children('div.default.text');
+
+  // Check <dropdown>'s placeholder.
+  assert.strictEqual(
+    Ember.$.trim($dropdownText.text()), placeholder);
+
+  // Change placeholder's value & check <dropdown>'s placeholder again.
+  placeholder = 'dropdown has no value';
+  this.set('placeholder', placeholder);
+  assert.strictEqual(Ember.$.trim($dropdownText.text()), placeholder);
+});
+
+test('readonly mode works properly', function(assert) {
+  assert.expect(4);
+
+  // Render component.
+  this.render(hbs`{{flexberry-dropdown
+    class=class
+  }}`);
+
+  // Retrieve component.
+  let $component = this.$().children();
+  let $dropdownMenu = $component.children('div.menu');
+
+  // Check that readonly (disabled) attribute doesn't exist yet.
+  assert.strictEqual($component.hasClass('disabled'), false, 'Component\'s hasn\'t readonly');
+
+  // Activate readonly mode & check that readonly (disabled) attribute exists now & has value equals to 'readonly'.
+  this.set('class', 'disabled');
+  assert.strictEqual($component.hasClass('disabled'), true, 'Component\'s has readonly');
+
+  // Check that readonly (disabled) attribute doesn't exist now.
+  this.set('class', '');
+  assert.strictEqual($component.hasClass('disabled'), false, 'Component\'s hasn\'t readonly');
+
+  // Check that component is disabled.
+  new Ember.RSVP.Promise((resolve, reject) => {
+    Ember.run(() => {
+      $component.click();
+    });
+
+    Ember.run(() => {
+      let animation = assert.async();
```

Comment:
It wiill be better to rosolve/reject promise & use assert.async in promise 'finally' callback, and assert.strictEqual in promise 'then' callback.

`
// Check that component is disabled.
let halfOfDropdownAnimationProceed = new Ember.RSVP.Promise((resolve, reject) => {
  Ember.run(() => {
    $component.click();
  });

  Ember.run(() => {
    setTimeout(() => {
      resolve();
    }, animationDuration / 2);
  });
});

 var asynchronousCheckCompleted = assert.async();
halfOfDropdownAnimationProceed .then(() => {
  assert.strictEqual($dropdownMenu.hasClass('animating'), false, 'Component is not active');
}).finally(() => {
  asynchronousCheckCompleted ();
});
`